cmd/kubeadm/app/cmd/join.go

```diff
@@ -56,6 +56,9 @@ func NewCmdJoin(out io.Writer, s *kubeadmapi.KubeadmConfig) *cobra.Command {
 		&s.Secrets.GivenToken, "token", "",
 		"(required) Shared secret used to secure bootstrap. Must match the output of 'kubeadm init'",
 	)
+	cmd.PersistentFlags().DurationVar(&s.JoinFlags.RetryTimeout, "retry-timeout", 5,
```

Comment:
You are probably right, one conf flag less with a reasonable default sounds practically applicable.